File: week08/23/prob1.c
#include <stdio.h>

int max_ropes_greater_than_k(int K, int A[], int n){
    int count = 0;
    int length = 0;
    for (int i = 0; i < n; i++){
        length += A[i];
        if (length >= K){
            count ++;
            length = 0;
        }
    }
    return count;
}

int main(){
    int A[] = {1,2,3,4,1,1,3};
    int K = 4;
    int n = sizeof(A) / sizeof(A[0]);
    int result = max_ropes_greater_than_k(K, A, n);
    printf("The answer is: %d \n", result);
}

File: week08/23/prob1.py
# Tie Rope Problem (Greedy)
    # max_ropes_greater_than_k

def max_ropes_greater_than_k(K, A):
    count = 0
    length = 0
    for rope in A:
        length += rope # add length with the previous one.
        if length >= K:
            count += 1
            length = 0
    return count

# Driver Code
if __name__=="__main__":
    A = [1,2,3,4,1,1,3]
    K = 4
    print(max_ropes_greater_than_k(K, A))